src/public/modules/forms/admin/controllers/view-responses.client.controller.js

```diff
@@ -137,7 +152,9 @@ function ViewResponsesController(
     }
 
     vm.csvDownloading = true
-    Submissions.downloadEncryptedResponses(params, vm.encryptionKey.secretKey)
+    vm.awaitDownload = false
```

Comment:
to be honest even after reading through the code a couple of times, I'm not sure what `awaitDownload` is supposed to be doing. could you give this variable a more informative name and add some comments to explain?

also, since it is logical state as opposed to view state (since it's not used in view-responses.client.view.html), this should be `let awaitDownload = false` instead of `vm.awaitDownload = false`.

Reply:
thanks for this. awaitDownload was intended to prevent premature setting of vm.csvDownloading = false as the .finally() block would execute before .catch() was complete (because .catch() did not return a promise). Upon review, I've refactored this to wrap the contents of .catch() in a promise which seems to be a more correct way of doing things